server/src/main/java/org/spine3/server/entity/storage/ColumnTypeRegistry.java

```diff
@@ -0,0 +1,133 @@
+
+    private final ImmutableMap<Class, C> columnTypeMap;
+
+    private ColumnTypeRegistry(Builder<C> builder) {
+        this.columnTypeMap = ImmutableMap.copyOf(builder.columnTypeMap);
+    }
+
+    /**
+     * @return an empty registry, used for
+     * {@linkplain org.spine3.server.storage.memory.InMemoryStorageFactory in-memory storages} etc.
+     */
+    public static <C extends ColumnType> ColumnTypeRegistry<C> empty() {
+        return ColumnTypeRegistry.<C>newBuilder().build();
+    }
+
+    /**
+     * Retrieves the {@link ColumnType} for specified {@link Column}.
+     *
+     * <p>By default, this method returns the {@link ColumnType} for the
+     * {@linkplain Column column's} {@linkplain Column#getType() type}.
+     *
+     * <p>If the {@link ColumnType} was not found by the {@code class} of the {@linkplain Column},
+     * its superclasses are checked one by one until a {@link ColumnType} is found or until
+     * the look up reaches the class representing {@link Object}.
+     *
+     * <p>If no {@link ColumnType} is found, an {@link IllegalStateException} is thrown.
+     *
+     * @param field the {@link Column} to get type conversion strategy for
+     * @return the {@link ColumnType} for the given {@link Column}
+     */
+    public C get(Column<?> field) {
+        checkNotNull(field);
+
+        Class javaType = field.getType();
+        javaType = Primitives.wrap(javaType);
+        C type = null;
+
+        while (type == null && javaType != null) {
+            type = columnTypeMap.get(javaType);
+            javaType = javaType.getSuperclass();
+        }
+
+        checkState(type != null,
+                   "Could not find storage type for %s.",
+                   field.getType());
+        return type;
+    }
+
+    @VisibleForTesting
+    Map<Class, C> getColumnTypeMap() {
+        return Collections.unmodifiableMap(columnTypeMap);
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder() {
+        return new Builder<>();
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder(ColumnTypeRegistry<C> src) {
+        final Builder<C> builder = new Builder<>();
+        builder.columnTypeMap.putAll(src.columnTypeMap);
+        return builder;
+    }
+
+    public static class Builder<C extends ColumnType> {
+
+        private final Map<Class, C> columnTypeMap = new HashMap<>();
+
+        private Builder() {
+        }
+
+        /**
+         * Create a mapping between the Java class and the database
+         * {@linkplain ColumnType column type}.
+         *
+         * @param javaType   the Java class to map the value from
+         * @param columnType the the database {@linkplain ColumnType column type} to map value to;
+         *                   must be an instance of {@code <C>}
+         * @param <J>        the Java type
+         * @return self for call chaining
+         */
+        public <J> Builder<C> put(Class<J> javaType, ColumnType<J, ?, ?, ?> columnType) {
```

Comment:
Done.

In fact, the main constraint for the `ColumnTypeRegistry` is matching the `class` object to the `ColumnType` object which receives the same type as the input.
Current signature guarantees that the matching is accurate.